index.js

```diff
@@ -119,61 +70,65 @@ const setWindowToLogin = () => {
   mainWindow.center()
 }
 
-
 // Start
 logger.debug("Run index.js in '" + MODE + "' mode")
 
 app.on('ready', () => {
   try {
     mainWindow = new BrowserWindow(connectWindowSize)
-    mainWindow.webContents.session.setDownloadPath(path.resolve(userDownloadDir))
+    mainWindow.webContents.session.setDownloadPath(OrbitConfig.userDownloadPath)
     Menu.setApplicationMenu(menu)
 
     // Pass the mode and electron flag to the html (renderer process)
     global.DEV = MODE === 'dev'
     global.isElectron = true
+    global.ipfsDaemonSettings = IpfsDaemonConfig
+
+    // Load the dist build or connect to webpack-dev-server
+    const indexUrl = MODE === 'dev'
+      ? 'http://localhost:8000/'
+      : 'file://' + __dirname + '/client/dist/index.html'
 
+    mainWindow.loadURL(indexUrl)
+
+    logger.info("started")
     // Resize the window as per app state
     ipcMain.on('connected', (event) => setWindowToNormal())
     ipcMain.on('disconnected', (event) => setWindowToLogin())
 
-    // Load and display a loading screen while we boot up
-    mainWindow.loadURL('file://' + __dirname + '/client/dist/loading.html')
-
-    // Bind the Orbit IPFS daemon to a random port, set CORS
-    IpfsDaemon(daemonSettings)
-      .then((res) => {
-        // We have a running IPFS daemon
-        const ipfsDaemon = res.daemon
-        const gatewayAddr = res.Addresses.Gateway
+    ipcMain.on('ipfs-daemon-start', (event, ipfsDaemonSettings) => {
+      logger.info(ipfsDaemonSettings)
+      // Bind the Orbit IPFS daemon to a random port, set CORS
+      IpfsDaemon(IpfsDaemonConfig)
```

Comment:
Done. Now the ipfs-daemon.config.js belongs to the client and takes the IpfsDataDir as argument (I think it's probably better to put it there since js-ipfs might use it as well...)